rational/src/lib.rs

```diff
@@ -774,6 +932,15 @@ mod test {
         let _a = Ratio::new(1, 0);
     }
 
+    #[test]
+    fn test_approximate_float() {
+        assert_eq!(super::approximate_float(0.5f32, 1.0e-20, 30), Some(Ratio::new(1i64, 2)));
+        assert_eq!(super::approximate_float(0.5f64, 1.0e-20, 30), Some(Ratio::new(1i32, 2)));
+        assert_eq!(super::approximate_float(5f32, 1.0e-20, 30), Some(Ratio::new(5i64, 1)));
+        assert_eq!(super::approximate_float(5f64, 1.0e-20, 30), Some(Ratio::new(5i32, 1)));
+        assert_eq!(super::approximate_float(29.97f32, 1.0e-20, 30), Some(Ratio::new(2997i64, 100)));
+        assert_eq!(super::approximate_float(-29.97f32, 1.0e-20, 30), Some(Ratio::new(-2997i64, 100)));
```

Comment:
Let me know if you want more tests. Might make sense to add something around quickcheck here at some point.